libopae/src/error_int.h

```diff
@@ -0,0 +1,52 @@
+// Copyright(c) 2018, Intel Corporation
+//
+// Redistribution  and  use  in source  and  binary  forms,  with  or  without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of  source code  must retain the  above copyright notice,
+//   this list of conditions and the following disclaimer.
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifndef __FPGA_ERROR_INT_H__
+#define __FPGA_ERROR_INT_H__
+
+#include <opae/types.h>
+
+#include "sysfs_int.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct error_list {
+	struct fpga_error_info info;
+	struct error_list *next;
+	char error_file[SYSFS_PATH_MAX];
+	char clear_file[SYSFS_PATH_MAX];
+};
```

Comment:
Have you thought about wrapping this in an outer struct with metadata (like the error count)?
It could look something like (and maybe more interesting metadata):
```C
struct error_info {
        struct error_list *errors;
        size_t count;
};
```

Reply:
`error_info` is the structure handled (and accessed directly) by users of the API. I'd rather not expose implementation-specific information (like sysfs paths) to them, since they'll start relying on that and it will break as soon as (for example) there's not sysfs entry for an error.